src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -0,0 +1,131 @@
+import CCPInitiationStrategyInterface from "./CCPInitiationStrategyInterface";
+import {hitch, isCitrixWebRTCSupported} from "../utils";
+import {FailedState} from "../rtc_session";
+import {RTC_ERRORS} from "../rtc_const";
+import "@citrix/ucsdk/CitrixWebRTC";
+
+export default class CitrixVDIStrategy extends CCPInitiationStrategyInterface {
+    constructor() {
+        super();
+        if (!isCitrixWebRTCSupported()) {
+            throw new Error('Citrix WebRTC redirection feature is NOT supported!');
+        }
+        window.getCitrixWebrtcRedir = function() {
+            const registryValue = Promise.resolve(1);
```

Comment:
whats with Promise.resolve(1)?

Reply:
Citrix SDK needs to retrieve redirection status registry value. We need to define a method on window object that returns this status registry value, and  the return value needs to be a Promise of value 0, 1, or empty string.
We define Promise.resolve(1) here to indicate the redirection status is success.